kroxylicious/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -137,37 +302,65 @@ public void channelRead(final ChannelHandlerContext ctx, Object msg) {
             outboundChannel.writeAndFlush(msg, outboundCtx.voidPromise());
             pendingFlushes = false;
         }
+        LOGGER.trace("/READ");
+    }
+
+    /**
+     * Sends an ApiVersions response from this handler to the client
+     * (i.e. prior to having backend connection)
+     */
+    private void writeApiVersionsResponse(ChannelHandlerContext ctx, DecodedRequestFrame<ApiVersionsRequestData> frame) {
+        // TODO check the format of the strings using a regex
```

Comment:
curiosity: why's the format of clientSoftwareName and  clientSoftwareVersion  so important we'd check them with a regex?  would we fail if they didnt match.

Reply:
It's to reproduce the broker behaviour. See org.apache.kafka.common.requests.ApiVersionsRequest#isValid. Added comment too.